exchanges/coinbene/coinbene_websocket.go

```diff
@@ -0,0 +1,328 @@
+package coinbene
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/orderbook"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/websocket/wshandler"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/websocket/wsorderbook"
+)
+
+const (
+	coinbeneWsURL = "wss://ws-contract.coinbene.vip/openapi/ws"
+	event         = "event"
+	topic         = "topic"
+)
+
+// WsConnect connects to websocket
+func (c *Coinbene) WsConnect() error {
+	if !c.Websocket.IsEnabled() || !c.IsEnabled() {
+		return errors.New(wshandler.WebsocketNotEnabled)
+	}
+	var dialer websocket.Dialer
+	err := c.WebsocketConn.Dial(&dialer, http.Header{})
+	if err != nil {
+		return err
+	}
+	go c.WsDataHandler()
+	if c.GetAuthenticatedAPISupport(exchange.WebsocketAuthentication) {
+		c.Login()
```

Comment:
Need to check the err message of this func and handle it (if json.Marshal fails or if it fails to send the websocket msg).